includes/content-distribution/class-incoming-post.php

```diff
@@ -173,6 +182,31 @@ protected function get_post_payload() {
 		return get_post_meta( $this->ID, self::PAYLOAD_META, true );
 	}
 
+	/**
+	 * Get payload from partial.
+	 *
+	 * @param array $payload The partial payload.
+	 *
+	 * @return array|WP_Error The full payload or WP_Error on failure.
+	 */
+	protected function get_payload_from_partial( $payload ) {
+		if ( ! $this->ID ) {
+			return new WP_Error( 'missing_post', __( 'Partial payload requires an existing post.', 'newspack-network' ) );
+		}
+
+		$current_payload       = $this->get_post_payload();
+		$current_payload_error = self::get_payload_error( $current_payload );
+		if ( is_wp_error( $current_payload_error ) ) {
+			return $current_payload_error->get_error_message();
```

Comment:
The error should be returned here I think –not a string.

Reply:
Good catch! Fixed in ac91081baaf6d1312071dbdddb85e8644e7a78c7